app/models/merchant.rb

```diff
@@ -32,6 +30,16 @@ def self.search_all(params)
     end
   end
 
+  def self.revenue(params)
+    revenue = Merchant.joins(invoices: [:transactions, :invoice_items]).
+      where(transactions: { result: "success" }).
+      where(id: params[:id]).
+      group("merchants.id").
+      sum("invoice_items.unit_price * invoice_items.quantity").
+      values.
+      first
 
+    { revenue: revenue / 100.0 }
```

Comment:
This can be extracted to be more SRP; maybe a 'revenue_format' method of some kind?